test/tests/query_params_test.js

```diff
@@ -3,8 +3,24 @@ import Router from "router";
 import { resolve, configure, reject, Promise } from "rsvp";
 
 var router, url, handlers, expectedUrl, actions;
+var scenarios = [
+  {
+    name: 'Sync Get Handler',
+    getHandler: function(name) {
+      return handlers[name] || (handlers[name] = {});
+    }
+  },
+  {
+    name: 'Async Get Handler',
+    getHandler: function(name) {
+      return Promise.resolve(handlers[name] || (handlers[name] = {}));
```

Comment:
I agree that I am just being paranoid, and your original code is certainly correct. Thank you for humoring me though.